### ISSUE TITLE
Kotlin 2.3 and experimental warnings?

### ISSUE THREAD

**Issue by tempacc21** [bazelbuild/rules_kotlin#1441]
Hey! 

I'm trying to use rules_kotlin v2.2.2 and kotlin 2.3, but I'm getting this warning: 

`warning: language version 2.3 is experimental, there are no backwards compatibility guarantees for new language and library features`

```
define_kt_toolchain(
    name = "kotlin_toolchain",
    api_version = "2.3",
    experimental_use_abi_jars = True,
    jvm_target = "21",
    kotlinc_options = ":kotlin_compiler_options",
    language_version = "2.3",
)
```
This doesnt play well with: 
```
kt_kotlinc_options(
    name = "kotlin_compiler_options",
    warn = "error",
    x_annotation_default_target = "first-only",
)

```

2.3 shouldn't be experimental, right? ref https://github.com/bazelbuild/rules_kotlin/pull/1410
